sycl/include/sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp

```diff
@@ -97,8 +132,23 @@ template <typename T, typename... Props>
 class __SYCL_SPECIAL_CLASS
 __SYCL_TYPE(annotated_ptr) annotated_ptr<T, detail::properties_t<Props...>> {
   using property_list_t = detail::properties_t<Props...>;
-  using reference =
-      sycl::ext::oneapi::experimental::annotated_ref<T, property_list_t>;
+
+  // buffer_location and alignment are allowed for annotated_ref
+  using allowed_properties =
+      std::tuple<decltype(buffer_location<0>), decltype(alignment<0>)>;
+  using filtered_properties =
+      typename PropertiesFilter<allowed_properties, Props...>::tuple;
+
+  // template unpack helper
+  template <typename... FilteredProps> struct unpack {};
+
+  template <typename... FilteredProps>
+  struct unpack<std::tuple<FilteredProps...>> {
+    using type = detail::properties_t<FilteredProps...>;
+  };
+
+  using reference = sycl::ext::oneapi::experimental::annotated_ref<
+      T, typename unpack<filtered_properties>::type>;
```

Comment:
According to the extension document the reference type should contain all properties. If we need to make this change we need to update the document too.

But why would we want to make this change? I understand why we need to filter before passing to the builtin but why already filter here?